### PR TITLE
creates EcoEngineAssembly

### PR DESCRIPTION
**Pull request by Hodge-zf**

### CODE REVIEW

**Comment by KenanJones** [comment]
I have a merge conflict. That is bad. I think that if you delete SportEngineAssembly.java it will go away.

**Comment by Hodge-zf** [comment]
Alright, I removed the file in conflict from my own repo, see what it says about the merge. I'm gonna try to finish up my slice within the next hour(s), too.

**Comment by Hodge-zf** [comment]
I noticed that on the UML diagram we are referring to some objects as Induction and some as Intake, and in the code so far some objects use induction and some use intake, as well as method names. Should we use one or the other to avoid confusion?

**Comment by KenanJones** [comment]
There is no longer a merge conflict, thanks.
We should probably use only induction, not intake. In your code, just use Induction, and we can fix the shared code tomorrow in class to prevent merge conflicts.

**Comment by KenanJones** [comment]
Your code looks good, just add unit tests for each instantiable class, and I can merge it in.

File: edu/jalc/automobile/onlinebuilder/components/engine/economy/EcoEngineAssembly.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  {
+   super();
+   this.exhaust = exhaust;
+   this.engine = engine;
+  }
+
+  public EconomyExhaust getExhaust()
+  {
+    return this.exhaust;
+  }
```
Comment (KenanJones) [inline edu/jalc/automobile/onlinebuilder/components/engine/economy/EcoEngineAssembly.java]: where is getEngine?

File: edu/jalc/automobile/onlinebuilder/components/engine/economy/boosted/supercharged/SuperchargedEcoEngine.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package edu.jalc.automobile.onlinebuilder.components.engine.economy.boosted.supercharged;
+
+import edu.jalc.automobile.parts.induction.SuperchargedInduction;
+
+abstract public class SuperchargedEcoEngine extends BoostedEcoEngine{
+  private SuperchargedInduction intake;
+}
```
Comment (KenanJones) [inline edu/jalc/automobile/onlinebuilder/components/engine/economy/boosted/supercharged/SuperchargedEcoEngine.java]: you need a constructor.

File: edu/jalc/automobile/onlinebuilder/components/engine/economy/boosted/turbocharged/TurbochargedEcoEngine.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package edu.jalc.automobile.onlinebuilder.components.engine.economy.boosted.turbocharged;
+
+import edu.jalc.automobile.parts.induction.TurbochargedInduction;
+
+abstract public class TurbochargedEcoEngine extends BoostedEcoEngine{
+  private TurbochargedInduction intake;
```
Comment (KenanJones) [inline edu/jalc/automobile/onlinebuilder/components/engine/economy/boosted/turbocharged/TurbochargedEcoEngine.java]: constructor needed here too.

File: edu/jalc/automobile/onlinebuilder/components/engine/economy/standard/StandardEcoEngine.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+   super(engine,exhaust);
+   this.intake = intake;
+  }
+
+ public NaturallyAspiratedInduction getIntake()
+ {
+   return this.intake;
+ }
+
+  public void getIntake(NaturallyAspiratedInduction intake)
```
Comment (KenanJones) [inline edu/jalc/automobile/onlinebuilder/components/engine/economy/standard/StandardEcoEngine.java]: this should be setIntake :)

**Pull request merged**